### PR TITLE
XMPP Notifier

### PR DESCRIPTION
**Pull request by depassp**
Added an XMPP notifier.  Works well with my Android phone (using google talk)

### CODE REVIEW

**Comment by depassp** [comment]
Here is a screenshot of what it looks like:

Screenshot

File: sickbeard/notifiers/xmpp.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+import sickbeard
+import lib.xmpp as xmpp
+
+from sickbeard import logger
+from sickbeard import common
+
+class XMPPNotifier:
+
+    def notify_download(self, ep_name):
+        if sickbeard.USE_XMPP:
```
Comment (lad1337) [inline sickbeard/notifiers/xmpp.py]: please use 

``` python
if sickbeard.USE_XMPP and sickbeard.XMPP_NOTIFY_ONSNATCH:
```

and appropriate in line 33

**Comment by lad1337** [comment]
please check for python2.5 compatibility
```sickbeard/notifiers/xmpp.py", line 65
    except Exception as inst:

``````

and please make the library stop printing to stdout
because i get stuff like that
```DEBUG: dispatcher   warn  Registering protocol "success" as <class 'lib.xmpp.protocol.Protocol'>(urn:ietf:params:xml:ns:xmpp-sasl)
``````

in the console ... either reroute to sickbeards logger or something else

and i do not seam to get download notifications although the sb log says
```[XMPP] Sending message 'Sick Beard Download Finished Dexter.6x02.Once.Upon.a.Time....SD.TV' to testreciver@googlemail.com

``````
also all your logging seam are in INFO lvl please use

```python
logger.log("to log",logger.DEBUG) # debug lvl
logger.log("to log",logger.ERROR) # error lvl
logger.log("to log",logger.WARNING) # warning lvl
``````

and in see my comment about snatch and download config

**Comment by depassp** [comment]
Changes made.  Is the typo in testreciver@googlemail.com intentional?

**Comment by depassp** [comment]
Screenshot of messages being received:

Screenshot

**Comment by kamu** [comment]
I like this. XMPP is great. Nice work.

File: sickbeard/notifiers/xmpp.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+             
+        return self._sendMessage(recipient, message='Test message from Sick Beard')
+        
+    def _sendMessage(self, recipient, message='Test message from Sick Beard'):
+        result = self._connect()
+        if result: return result
+        #Send Message
+        logger.log("[XMPP] Sending message '" + message + "' to recipient " + recipient, logger.DEBUG)
+        try:
+            self.cnx.send(xmpp.Message(recipient, message))
```
Comment (midgetspy) [inline sickbeard/notifiers/xmpp.py:111]: This doesn't exist, throws an exception.

File: sickbeard/notifiers/xmpp.py [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+            logger.log("[XMPP] Authentication Failed", logger.ERROR)
+            return "Failed to authenticate. Please make sure the username & password are correct."
+
+    def notify_download(self, ep_name):
+        if sickbeard.USE_XMPP and sickbeard.XMPP_NOTIFY_ONDOWNLOAD:
+            return self._sendMessage(message=' '.join(['Sick Beard', common.notifyStrings[common.NOTIFY_DOWNLOAD], ep_name]))
+    
+    def notify_snatch(self, ep_name):
+        if sickbeard.USE_XMPP and sickbeard.XMPP_NOTIFY_ONSNATCH:
+            return self._sendMessage(message=' '.join(['Sick Beard', common.notifyStrings[common.NOTIFY_SNATCH], ep_name]))
```
Comment (midgetspy) [inline sickbeard/notifiers/xmpp.py:81]: Why do these have no recipient?

**Comment by midgetspy** [comment]
This request doesn't merge as it is, I've fixed it up in a branch called xmpp. Once the merge is finished though it doesn't actually work - there are a couple problems with the notifier itself. If you are going to resubmit this just fix it based on my xmpp branch.

**Pull request closed without merging**